src/lsdb/catalog/catalog.py

```diff
@@ -68,3 +80,91 @@ def get_partition_index(self, order: int, pixel: int) -> int:
             raise ValueError(f"Pixel at order {order} pixel {pixel} not in Catalog")
         partition_index = self._ddf_pixel_map[hp_pixel]
         return partition_index
+
+    @property
+    def name(self):
+        """The name of the catalog"""
+        return self.hc_structure.catalog_name
+
+    def crossmatch(
+        self,
+        other: Catalog,
+        suffixes: Tuple[str, str] | None = None,
+        algorithm: Type[AbstractCrossmatchAlgorithm]
+        | BuiltInCrossmatchAlgorithm = BuiltInCrossmatchAlgorithm.KD_TREE,
+        name: str | None = None,
+        **kwargs,
+    ) -> Catalog:
+        """Perform a cross-match between two catalogs
+
+        The pixels from each catalog are aligned via a `PixelAlignment`, and cross-matching is
+        performed on each pair of overlapping pixels. The resulting catalog will have partitions
+        matching an inner pixel alignment - using pixels that have overlap in both input catalogs
+        and taking the smallest of any overlapping pixels.
+
+        The resulting catalog will be partitioned using the left catalog's ra and dec, and the
+        index for each row will be the same as the index from the corresponding row in the left
+        catalog's index.
+
+        Args:
+            other (Catalog): The right catalog to cross-match against
+            suffixes (Tuple[str, str]): A pair of suffixes to be appended to the end of each column
+                name when they are joined. Default: uses the name of the catalog for the suffix
+            algorithm (BuiltInCrossmatchAlgorithm | ufunc): The algorithm to use to perform the
+                crossmatch. Can be either a string to specify one of the built-in cross-matching
+                methods, or a custom method defined by subclassing AbstractCrossmatchAlgorithm.
+
+                Built-in methods:
+                    -`kd_tree`: find the k-nearest neighbors using a kd_tree
+
+                Custom function:
+                    To specify a custom function, write a class that subclasses the
+                    `AbstractCrossmatchAlgorithm` class, and overwrite the `crossmatch` function.
+
+                    The function should be able to perform a crossmatch on two pandas DataFrames
```

Comment:
This level of documentation is making me happy.

Let's twiddle it to work well with autoapi rendering: https://lsdb--24.org.readthedocs.build/en/24/autoapi/lsdb/catalog/catalog/index.html#lsdb.catalog.catalog.Catalog.crossmatch

Reply:
Wooah I didn't know it could do that for PRs, that's cool

Reply:
Pretty sure this is rendered ok now, any other tweaks you were thinking about?